enhancements/machine-config/bootc_update_path.md

```diff
@@ -0,0 +1,185 @@
+    - Why non-goal: This is a valid request and will help clear up MCO’s update and enhance efficiency. This is non-goal for this enhancement because the update consolidation effort shall take place after a TP implementation of bootc switch and shall be followed up in a separate epic. 
+* Bootc Day-2 Configuration Tools 
+    - Description: Bootc has opened a door for disk image customization via configmap. Switching from rpm-ostree to bootc, MCO should not only make sure all the functionality remains but also proactively extending its support to make sure all the customization power brought in by bootc are directed to the user, allowing them to fully maximize these advantages.This will involve creating a new user-side API for fetching admin-defined configuration and pairing MCO with a bootc day-2 configuration tool for applying the customizations. Interest and Proposals in which has been raised several times in previous discussions.
+    - Why non-goal: This is a valid request and will enhance user experience by increasing customization option.This is a non-goal for this enhancement because this request should be the next step after bootc switch happening. It will be announced and discussed in detail in a follow-up enhancement. 
+* Splitting the RHCOS Image into layers will be announced in the other enhancement
+
+## Proposal
+
+MCO allows the users to specify changes in osImageURL Update / Extension Change / Kernel Argument Update / Kernel Type Switch and land the changes by selling out various rpm-ostree commands. With this enhancement, the image update path in the MCO will be further extended by adding support for bootc, which is already included in the RHCOS images. The new update path will be merged behind a featuregate, which will be removed once the feature becomes GA (graduation criteria discussed in Section Graduation Criteria) The goal is to ensure that all the functionality remains when we rip off the rpm-ostree backbone and have bootc in place. This aligns with our ultimate goal: Have users not notice any difference in user experiences other than that more customization and mutation options are supported. Though we are working towards maintaining as much functionality as possible, several known limitations are noticed and discussed in the workflow for the non-On-Cluster-Layering and On Cluster Layering path below.
+
+Without On Cluster Layering: 
```

Comment:
Based on my reading of this section, it's implying that non-on cluster layering will never use bootc 

We can maybe move the limitations to a later section and clarify that plan here?